src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -683,23 +678,31 @@ def create(exported_python_target):
 
         interpreter = self.context.products.get_data(PythonInterpreter)
         python_dists = self.context.products.register_data(self.PYTHON_DISTS_PRODUCT, {})
+
+        setup_runner = SetupPyRunner.Factory.create(
+            scope=self,
+            interpreter=interpreter,
+            pex_file_path=os.path.join(self.workdir, self.fingerprint, "setup-py-runner.pex"),
+        )
         for exported_python_target in reversed(sort_targets(list(created.keys()))):
             setup_dir = created.get(exported_python_target)
             if setup_dir:
                 if not self._run:
-                    self.context.log.info("Running packager against {}".format(setup_dir))
-                    setup_runner = Packager(setup_dir, interpreter=interpreter)
-                    tgz_name = os.path.basename(setup_runner.sdist())
+                    self.context.log.info("Running sdist against {}".format(setup_dir))
+                    sdist = setup_runner.sdist(setup_dir)
+                    tgz_name = sdist.name
                     sdist_path = os.path.join(dist_dir, tgz_name)
                     self.context.log.info("Writing {}".format(sdist_path))
-                    shutil.move(setup_runner.sdist(), sdist_path)
-                    safe_rmtree(setup_dir)
+                    shutil.move(sdist, sdist_path)
+                    safe_rmtree(str(setup_dir))
```

Comment:
@codealchemy pointed out that it's better to use `Path.as_posix()` than `str(Path)` because it better expresses intent. They behave identically, except that `str(Path)` does the correct thing on Windows and `Path.as_posix()` obviously doesn't, but we don't support Windows.

Reply:
I'm not sure I agree that `Path.as_posix()` better expresses intent. Doing so would embed the knowledge that "we don't support Windows" here. `str(instance of Path)` seems pretty idiomatic, and expresses our true intent, which is "turn this Path into a string that is valid on this platform".

Reply:
> and expresses our true intent, which is "turn this Path into a string that is valid on this platform"

And really, our intent is not even that, its more accurately just to convert between stdlib APIs that the Python team didn't get around to. Ideally we'd just `setup_dir.rmtree()` and remain ignorant of how to represent the `Path` to a legacy API.